chalicelib/decorators.py

```diff
@@ -7,4 +25,52 @@ def wrapper(self, table_name: str, *args, **kwargs):
 
         return func(self, table_name, *args, **kwargs)
 
-    return wrapper
\ No newline at end of file
+    return wrapper
+
+
+def auth(blueprint, roles):
+    """
+    Decorator for authenticating and authorizing access to API routes.
+
+    Args:
+        blueprint (object): The Chalice Blueprint object, providing access to the current request.
+        role (list[str]): The required role for authorization.
```

Comment:
u spelled it wrong...

Reply:
i did not know decorator functions was a thing ... very cool and useful 👍

Reply:
should be roles 🥲